route/v1/types.go

```diff
@@ -369,14 +369,15 @@ type RouteIngressConditionType string
 const (
 	// RouteAdmitted means the route is able to service requests for the provided Host
 	RouteAdmitted RouteIngressConditionType = "Admitted"
-	// TODO: add other route condition types
+	// RouteUpgradable indicates the route is upgradeable to the next version of OpenShift
```

Comment:
please describe what happens if this is false.  Is the upgrade stopped?  That's a lot of power to give a namespace editor compared a cluster-admin assessing a CVE for instance.

If the upgrade isn't stopped, how do we expect a cluster-admin to avoid accidentally causing this situation?

cc @sdodson 

for discussion of how to handle upgradeability.

Reply:
> please describe what happens if this is false. Is the upgrade stopped? That's a lot of power to give a namespace editor compared a cluster-admin assessing a CVE for instance.

I will update code comment soon - the cluster ingress operator will read the route's Upgradeable status and set the cluster operator ingress Upgradeable status which will prevent upgrades. So yes, the route owner can stop an upgrade, agreed.

See https://redhat-internal.slack.com/archives/CEGKQ43CP/p1704894930666179 discussion.